core/src/main/java/org/pentaho/platform/util/ActionUtil.java

```diff
@@ -210,11 +212,53 @@ public static String generateWorkItemUid( final Map<String, Serializable> params
     // corresponding action keys (see KEY_MAP), fall back on the quartz key, if the action key cannot be found
     final String userName = Optional.ofNullable( (String) params.get( INVOKER_ACTIONUSER ) ).orElse( (String) params
       .get( QUARTZ_ACTIONUSER ) );
-    final String inputFilePath = Optional.ofNullable( (String) params.get( INVOKER_STREAMPROVIDER_INPUT_FILE ) )
+    String inputFilePath = Optional.ofNullable( (String) params.get( INVOKER_STREAMPROVIDER_INPUT_FILE ) )
       .orElse( (String) params.get( QUARTZ_STREAMPROVIDER_INPUT_FILE ) );
+
+    // if we still cannot find the inputFile, we have one last attempt: check for inline passing of input and output files,
+    // represented by 'ActionAdapterQuartzJob-StreamProvider -> input file = <path><colon>output file=<path>
+    if ( StringUtil.isEmpty( inputFilePath ) && isInlinePassingOfInputOnStreamProvider( params ) ) {
+      inputFilePath = getInlineInputFileOnStreamProvider( params );
+    }
+
     return generateWorkItemUid( inputFilePath, userName );
   }
 
+  public static boolean isInlinePassingOfInputOnStreamProvider( final Map<String, Serializable> params ) {
+
+    if ( params != null ) {
+
+      String streamProviderContent = ( params.containsKey( QUARTZ_STREAMPROVIDER )
+              ? params.get( QUARTZ_STREAMPROVIDER ) : params.containsKey( INVOKER_STREAMPROVIDER )
+              ? params.get( INVOKER_STREAMPROVIDER ) : StringUtils.EMPTY ).toString().trim();
+
+      return streamProviderContent.contains( QUARTZ_STREAMPROVIDER_INLINE_INPUT_FILE );
+    }
+
+    return false;
+  }
+
+  public static String getInlineInputFileOnStreamProvider( final Map<String, Serializable> params ) {
+
+    if ( isInlinePassingOfInputOnStreamProvider( params ) ) {
+
+      try {
+
+        String streamProviderContent = ( params.containsKey( QUARTZ_STREAMPROVIDER )
+                ? params.get( QUARTZ_STREAMPROVIDER ) : params.containsKey( INVOKER_STREAMPROVIDER )
+                ? params.get( INVOKER_STREAMPROVIDER ) : StringUtils.EMPTY ).toString().trim();
+
```

Comment:
By now, we have done the same thing twice, as in to check if the params conatins key `QUARTZ_STREAMPROVIDER`, right?

Reply:
correct; while `isInlinePassingOfInputOnStreamProvider` is a predicate alone, `getInlineInputFileOnStreamProvider` does provide content; I felt it would be safer to wrap the `get` over that predicate check, because they are 2 separate `public static` methods; so this means that they can be called independently.

Because the predicate check is very light & quick, I felt the added safety-check is more beneficial than the eventual possibility of the check being done twice.

I can change this to **not** have the `get` call for the predicate. If you agree, then I could tackle this in a follow-up "Cleanup" PR, rather than blocking this one from getting processed.

Thoughts ?

Reply:
Sounds good!